samanthas_telegram_bot/api_queries/smalltalk.py

```diff
@@ -0,0 +1,205 @@
+            )
+            user_data.comment = (
+                f"{user_data.comment}\n- SmallTalk assessment not finished\nCheck {result.url}"
+            )
+            return None
+        elif result.status == SmalltalkTestStatus.RESULTS_NOT_READY:
+            if attempts > 10:
+                await log_and_report(
+                    bot=update.get_bot(),
+                    logger=logger,
+                    level="error",
+                    text=(
+                        f"Chat {user_data.chat_id}: SmallTalk results for {user_data.first_name} "
+                        f"{user_data.last_name} still not ready after 5 minutes. "
+                        f"Interview ID {user_data.student_smalltalk_test_id}."
+                    ),
+                    parse_mode=None,
+                )
+                user_data.comment = (
+                    f"{user_data.comment}\n- SmallTalk assessment results were not ready\n"
+                    f"Interview ID {user_data.student_smalltalk_test_id}"
+                )
+
+            logger.info(f"Chat {user_data.chat_id}: SmallTalk results not ready. Waiting...")
+            attempts += 1
+            await asyncio.sleep(30)
+        else:
+            await log_and_report(
+                bot=update.get_bot(),
+                logger=logger,
+                level="info",
+                text=(
+                    f"Chat {user_data.chat_id}: Received [SmallTalk results for "
+                    f"{user_data.first_name} {user_data.last_name}]({result.url})"
+                ),
+                parse_mode=ParseMode.MARKDOWN_V2,
+            )
+            return result
+
+
+async def get_json_with_results(test_id: str) -> bytes:
+    async with httpx.AsyncClient() as client:
+        r = await client.get(
+            url=f"{URL_PREFIX}/test_status",
+            headers=HEADERS,
+            params={
+                "id": test_id,
+                "additional_fields": (
+                    "detailed_scores,strength_weaknesses,problem_statuses,problem_titles"
+                ),
+            },
+        )
+
+    logger.info(
+        f"Request headers: {r.request.headers}. "
+        f"Response: {r.status_code=}, {r.headers=}, {r.content=}"
+    )
+    return r.content
+
+
+def process_smalltalk_json(json_data: bytes) -> SmalltalkResult | None:
+    def level_is_undefined(str_: str) -> bool:
+        return str_.lower().strip() == "undefined"
+
+    try:
+        loaded_data = json.loads(json_data)
+    except json.decoder.JSONDecodeError:
+        logger.error(f"Could not load JSON from {json_data=}")
+        return None
+
+    status = loaded_data["status"]
+
+    # TODO Don't want to raise NotImplementedError here, but think about it
+    if status not in SmalltalkTestStatus._value2member_map_:  # noqa
+        logger.warning(f"SmallTalk returned {status=} but we have no logic for it.")
+
+    if status == SmalltalkTestStatus.NOT_STARTED_OR_IN_PROGRESS:
```

Comment:
Maybe try to use switch-case here?

Reply:
IDK, it gives 5 lines instead of 4. But I replaced second `if` with `elif`. The third `if` is independent, it can't be put into a `switch`.

Reply:
Ok, let's keep it so